src/openforms/validations/registry.py

```diff
@@ -151,6 +96,12 @@ def validate(
         else:
             return ValidationResult(True)
 
+    def check_plugin(self, plugin: BasePlugin) -> None:
+        if not hasattr(plugin, "value_serializer"):
+            raise ValueError(
+                f"Validator '{plugin.identifier}' must have a 'value_serializer' attribute."
+            )
```

Comment:
is there a way we could enforce this via `@abc` perhaps instead of making a runtime check?

I'm also okay with changing this to an `assert ...` instead.

Reply:
This check can actually be removed if the behavior described in https://github.com/open-formulieren/open-forms/pull/3741#discussion_r1444796139 is fine. Otherwise, we could enforce it as an abstract property maybe

Reply:
I can't follow, what exactly is needed to remove this check?

Reply:
If we keep the default value for `value_serizlier`:

```python
value_serializer: ClassVar[type[serializers.BaseSerializer]] = StringValueSerializer
```

then all subclasses are guaranteed to have the attribute

Reply:
great, keep the default value.